### PR TITLE
Validate playground

### PR DESCRIPTION
**Pull request by Palleas**
This pull request addresses #2879.

I moved the command building the framework into a bash script, called "build". Should I go for something more explicit with "lint", "check"..? Also I'm not sure I should lint the playground with all the platform, I should probably do it only for the mac.

This is a WIP until the Travis build actually passes. 🍀

### CODE REVIEW

File: script/build [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      -sdk "$XCODE_SDK" \
+      -derivedDataPath "${BUILD_DIRECTORY}" \
+      -configuration $CONFIGURATION \
+      ENABLE_TESTABILITY=YES \
+      GCC_GENERATE_DEBUGGING_SYMBOLS=NO \
+      RUN_CLANG_STATIC_ANALYZER=NO \
+      | xcpretty
+
+# Compile code in playgrounds 
+# TODO only run for mac ?
```
Comment (RuiAAPeres) [inline script/build]: Since the playground is targeting Mac, yes.

**Comment by ikesyo** [comment]
It seems that this is failing:

```bash
script/validate-playground.sh: line 15: build/Build/Products/Release/all-playground-pages.swift: No such file or directory
Apple Swift version 2.2 (swiftlang-703.0.18.8 clang-703.0.31)
Target: x86_64-apple-macosx10.9
<unknown>:0: error: no such file or directory: 'build/Build/Products/Release/all-playground-pages.swift'
```

**Comment by Palleas** [comment]
The build seems to be passing now and I've squashed my numerous commits. Please let me know if I need to change anything else in this PR. 😸

File: script/build [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+if [[ -z $XCODE_ACTION ]]; then
+      echo "\$XCODE_ACTION is not set. Using \`build\`"
+      XCODE_ACTION="build"
+fi
+
+if [[ -z $XCODE_SDK ]]; then
+      echo "\$XCODE_SDK is not set. Using \`macosx\`"
+      XCODE_SDK="macosx"
+fi
```
Comment (mdiep) [inline script/build]: I'm not wild about duplicating these is in the script. Can we just require them to be set?

File: script/build [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      GCC_GENERATE_DEBUGGING_SYMBOLS=NO \
+      RUN_CLANG_STATIC_ANALYZER=NO \
+      | xcpretty
+
+# Compile code in playgrounds 
+if [[ $XCODE_SDK -eq "macosx" ]]; then
+      echo "SDK is $XCODE_SDK, validating playground..."
+      . script/validate-playground.sh
+else
+      echo "SDK is $XCODE_SDK, not validating playground."
```
Comment (mdiep) [inline script/build]: I don't think we need to echo in the else case.

**Comment by mdiep** [comment]
Thanks for talking this! I've left a few notes. ☺️ 

FWIW, you don't need to squash your commits.

**Comment by Palleas** [comment]
The script now fails if vars are not set and nothing is displayed if we don't validate the playground. Hopefully the build should still pass!

**Comment by mdiep** [comment]
Looks good to me. 👍

**Pull request merged**